tests/test_datamixing.py

```diff
@@ -383,3 +387,38 @@ def test_phase07_knowledge_samples_have_unmask_true():
     assert any(
         sample["unmask"] for sample in phase10_ds
     ), "No samples in phase10 have unmask=True"
+
+
+def test_mix_instructlab_07x_precomputed_skills_with_unmask(tmp_path):
+    """
+    Test that we can mix the precomputed skills data format used in
+    InstructLab 0.7.x with new data that has the unmask field.
+    """
+
+    # Create a knowledge dataset
+    knowledge_dataset = load_knowledge_dataset()
+
+    # Create phase07 dataset
+    phase10_ds = _create_phase10_ds(
+        generated_dataset=knowledge_dataset,
+        auxiliary_inst=auxiliary_inst,
+        use_legacy_pretraining_format=False,
+    )
+    phase10_path = os.path.join(tmp_path, "knowledge_p10.jsonl")
+    jldump(phase10_ds, phase10_path)
+
+    output_path = os.path.join(tmp_path, "output.jsonl")
+    recipe = Recipe()
+    # Add an old precomputed skills dataset in that does NOT have an unmask field
+    recipe.add_dataset(TEST_PRECOMPUTED_07X_PATH, 1.0)
+    # Add in our new phase10 dataset that does have unmask fields
+    recipe.add_dataset(phase10_path, 1.0)
+    # Mix the two datasets, ensuring nothing errors out
+    recipe.save_mixed_dataset(output_path, TEST_NUM_PROCS)
```

Comment:
@bbrowning I'm realizing that we should also perhaps test that when these two are combined like this and saved, loading them doesn't break either. 

Because I've hit situations where `datasets.concatenate_datasets` will work and save them, but then loading them breaks.

Reply:
Good idea - this appears to work fine, and I opened #600 to adjust the test to load the mixed dataset via `load_dataset` instead of `jlload` that I was using just to ensure this keeps working.

Reply:
Awesome, thank you so much Ben!